spec/test_config.rb

```diff
@@ -0,0 +1,42 @@
+require 'yaml'
+
+module TestConfig
+  class << self
+    def current_config
+      self[current_adapter]
+    end
+
+    def [](adapter_name)
+      config[adapter_name] || fail("the adapter '#{adapter_name}' is not configured")
+    end
+
+    def adapters
+      config.keys
+    end
+
+    def current_adapter
+      ENV.fetch('TEMPING_ADAPTER')
+    end
+
+    def current_adapter=(current_adapter)
+      ENV['TEMPING_ADAPTER'] = current_adapter
```

Comment:
When you call `rake` you start a `ruby` process (e.g. `ruby /home/grn/.rbenv/versions/2.1.5/bin/rake`). The spec tasks runs the `rspec` command which spawns a ruby child process (via `fork`/`exec`). This is clearly seen after inserting `system("pstree -ahl")` at the top of `spec_helper.rb` (an excerpt from the output):

```
  bash --login
    └─ruby /home/grn/.rbenv/versions/2.1.5/bin/rake
        ├─ruby -S rspec ./spec/temping_spec.rb --colour
        │   ├─pstree -ahl
        │   └─{ruby-timer-thr}
        └─{ruby-timer-thr}
```

A rake task, which resides in the parent process, chooses the adapter to test. However, the test itself is carried out by the subprocess. We need to send the information about which adapter to test from the parent process to the child process and an environment variable was the simplest choice.